src/main/java/com/example/SOPSbackend/security/BasicUserService.java

```diff
@@ -0,0 +1,21 @@
+package com.example.SOPSbackend.security;
+
+import com.example.SOPSbackend.model.BasicUser;
+import com.example.SOPSbackend.repository.UserRepository;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+public class BasicUserService<T extends BasicUser> implements UserDetailsService {
```

Comment:
Dlaczego "Basic"User? Chyba lepiej po prostu "User" to wygląda jakby był jakiś jeszcze inny User (rozumiem, że są lekarze, admini itd, ale to nie ma żadnego "Super"Usera - coś przegapiłem?). Po prostu User.

IMO

Reply:
User występuje w Springu i IDE będzie często myliło obie klasy i importowało złą.  Najwygodniej po prostu zmienić nazwę na lekko inną.

Reply:
Zupełnie się nie zgadzam (IDE importuje to o co się poprosi) nawet w jednym projekcie może być kilka klas o tej samej nazwie, nic złego się nie stanie. 

Ale nie musimy się wszędzie zgadzać, kliknij „resolved” jeżeli nie masz zamiaru zmieniać. Po tym można mergować.

Reply:
Btw, zabiliśmy kod o jakieś 25 procent. Super robota 🔥🔥🔥